### PR TITLE
RFC/DNM: Initial export to nextpnr

### PR DESCRIPTION
**Pull request by litghost**

### CODE REVIEW

File: fpga_interchange/populate_chip_info.py [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+                    wire_index=idx,
+                    site_index=site_index)
+
+            site_wire_index = self.add_wire(flat_wire)
+            for pin in site_wire.pins:
+                assert pin not in bel_pin_to_site_wire_index
+                bel_pin_to_site_wire_index[pin] = site_wire_index
+
+        # Add BELs
+        for bel_idx, bel in enumerate(site_type.bels):
```
Comment (daveshah1) [inline fpga_interchange/populate_chip_info.py:212]: I'm not sure if there is any need to import site port bels as bels; given pips are being inserted for the tile/site boundary already and are more suited for this kind of purpose inside nextpnr. They won't do any harm, but will potentially add clutter and marginally slow things down.

It's also arguable as to whether routing bels are actually needed as bels either - again importing these as just pips might be better - I know nextpnr-xilinx historically created bels for these but I don't remember if they were ever actually used as bels. All the needed information could probably be tagged to the pips.

Comment (litghost) [reply fpga_interchange/populate_chip_info.py:212]: > It's also arguable as to whether routing bels are actually needed as bels either - again importing these as just pips might be better - I know nextpnr-xilinx historically created bels for these but I don't remember if they were ever actually used as bels. All the needed information could probably be tagged to the pips.

I actually ended up using both the routing and site pin BELs, so for now they are getting emitted.  If it is a problem to have these BELs in the chip data, I could put the routing and site pin BELs at the end of the tile type bel_data array so they are easily skipped.

Comment (daveshah1) [reply fpga_interchange/populate_chip_info.py:212]: That's fair enough - the only real problems I was thinking were related to debugging and putting them at the end would make it a bit clearer what was going on

File: fpga_interchange/populate_chip_info.py [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+
+    # Output nodes
+    for idx, node in enumerate(device.device_resource_capnp.nodes):
+        # Skip nodes with only 1 wire!
+        if len(node.wires) == 1:
+            continue
+
+        node_info = NodeInfo()
+
+        # FIXME: Replace with actual node name?
```
Comment (daveshah1) [inline fpga_interchange/populate_chip_info.py:710]: Do we need to handle the case where the node name isn't one of the wire names inside the node? If not; either we could say the node name always comes from wire index 0; or store an index to the 'primary' wire for naming purposes

Comment (litghost) [reply fpga_interchange/populate_chip_info.py:710]: So I realized when debugging that RapidWright emits the node where the first wire in the node is the node name.  So I believe the bba as emitted will have the correct `getWireByName` that corresponds to the name of the node.

Comment (litghost) [reply fpga_interchange/populate_chip_info.py:710]: > either we could say the node name always comes from wire index 0

To be clear, this is what the code is doing right now.

**Pull request closed without merging**